### PR TITLE
Simply setting errorData | calling error messages

### PR DESCRIPTION
**Pull request by AlmightyMikkel**

### CODE REVIEW

**Comment by ladydascalie** [comment]
From a backend perspective, this is good :)

File: Runtime/Client/LootLockerServerRequest.cs [hunk 1]
```diff
@@ -174,8 +183,11 @@ public class LootLockerResponse
             {
                 return LootLockerResponseFactory.Error<T>("Unknown error, please check your internet connection.");
             }
-            else if (serverResponse.errorData != null && !string.IsNullOrEmpty(serverResponse.errorData.code))
+            else if (serverResponse.errorData != null)
             {
+                if(!string.IsNullOrEmpty(serverResponse.errorData.code)) {
```
Comment (kirre-bylund) [inline Runtime/Client/LootLockerServerRequest.cs:188]: The inside and outside of this if statement are the same, meaning it is pointless as it is now. Either move it back up or if you decide to go with the intended change in the rest of the code then this one is moot and the errorData != null is sufficient.

File: Runtime/Client/LootLockerServerRequest.cs [hunk 2]
```diff
@@ -116,6 +116,15 @@ public class LootLockerErrorData
...
         /// </summary>
         public string message { get; set; }
+
+        /// <summary>
+        /// An easy way of debugging LootLockerErrorData class, example: Debug.Log(onComplete.errorData);
+        /// </summary>
+        /// <returns>string used to debug errors</returns>
+        public override string ToString()
+        {
+            return $"We encountered an unexpected server error. Please try again later.\n If the issue persists, please contact LootLocker support and reference the following error details:\n trace ID - {trace_id},\n request ID - {request_id},\n message - {message}.";
```
Comment (kirre-bylund) [inline Runtime/Client/LootLockerServerRequest.cs:126]:
```suggestion
            return $"An unexpected server error occured, please try again later.\n If the issue persists, please contact LootLocker support and provide the following error details:\n trace ID - {trace_id},\n request ID - {request_id},\n message - {message}.";
```

I think generally having a ToString on errorData is fantastic. Can you make sure that it prints nicely in editor though? It's rather long, and we already have feedback that the important part of the error messages disappear out of the log line in Unity because of the length.

I also kinda object to this since "Please try again later" means that it is a problem on **our** end. If it's a 4xx error then they can retry all they want, it's not gonna change anything. So we probably should at least distinguish those cases.

File: Runtime/Client/LootLockerServerApi.cs [hunk 3]
```diff
@@ -131,14 +131,7 @@ IEnumerator coroutine()
...
 
-                    LootLockerErrorData errorData =
-                        LootLockerJson.DeserializeObject<LootLockerErrorData>(webRequest.downloadHandler.text);
-                    if (ErrorIsPreErrorCodes(errorData))
-                    {
-                        errorData = ExpandErrorInformation(webRequest.responseCode, webRequest.downloadHandler.text);
-                    }
-
-                    response.errorData = errorData;
+                    response.errorData = LootLockerJson.DeserializeObject<LootLockerErrorData>(webRequest.downloadHandler.text);
```
Comment (kirre-bylund) [inline Runtime/Client/LootLockerServerApi.cs:134]: This is the part I need explained to me before we go on with the change. If it does indeed work it's fantastic. So much cleaner. I know that I intentionally made it this cumbersome a couple of months ago because I hade edge cases where the respnse did not look like
```json
{
    "error_code": "asdasd",
    "error_doc_url": "asdkaskdl",
    ....
}
```
And if it doesn't then this will cause errorData to be null, which then will cause errorData to be null and the intended "helping" structure of errorData to be pointless. 

To summarize, I can approve this wholeheartedly as long as @ladydascalie can guarantee that **all** potential errors we can get (from Go and PHP, network issues, etc.) will be packaged in that beautiful structure. But I'll need to hear that case first.

**Review by kirre-bylund — CHANGES_REQUESTED** [review]
